speed_fan.h includes

### The problem

SpeedFan should not require output component anymore. 
> output (Optional, [ID](https://esphome.io/guides/configuration-types#config-id)): The id of the [float output](https://esphome.io/components/output/#output) to use for this fan.

Omitting the output leads to compilation errors due to missing headers, cause there is still an implicit dependency to output component.

### Which version of ESPHome has the issue?

ESPHome 2024.2.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2024.2.3

### What platform are you using?

ESP8266

### Board

NodeMCU v2

### Component causing the issue

speed

### Example YAML snippet

```yaml
esphome:
  name: test
  friendly_name: test

esp8266:
  board: nodemcuv2

# Enable logging
logger:

fan:
  - platform: speed
    name: my_fan
    id: my_fan
    speed_count: 3
    on_speed_set:
      - logger.log: "Fan Speed was changed!"
```


### Anything in the logs that might be useful for us?

```txt
In file included from src/esphome/components/speed/fan/speed_fan.cpp:1:
src/esphome/components/speed/fan/speed_fan.h:6:10: fatal error: esphome/components/output/binary_output.h: No such file or directory
    6 | #include "esphome/components/output/binary_output.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pioenvs/test/src/esphome/components/speed/fan/speed_fan.cpp.o] Error 1
========================= [FAILED] Took 28.12 seconds =========================
```
```


### Additional information

_No response_

--- a/esphome/components/speed/fan/__init__.py
+++ b/esphome/components/speed/fan/__init__.py
@@ -14,6 +14,8 @@ from esphome.const import (
 
 from .. import speed_ns
 
+AUTO_LOAD = ["output"]
+
 SpeedFan = speed_ns.class_("SpeedFan", cg.Component, fan.Fan)
 
 CONFIG_SCHEMA = fan.FAN_SCHEMA.extend(

